### ISSUE TITLE
Random Seed on LightGBM

### ISSUE THREAD

**Issue by cagataytuylu** [microsoft/LightGBM#4562]
## Description

I was working on demand forecasting in algorithm the other day. I was going over my daytime work and doing my vicar calculations of 30000 iterations. However, on the same day, because the code was taking too long, I stopped the code and turned off my computer and went home. When I got home and turned it on again, the algorithm started to stop after around 200 iterations. After the one-hot process after feature engineering, I recorded the yield ready for modeling as both pickle and parquet. When I go home in the evening, this problem exists even though I proceed through that data. I haven't made any changes to my codes. This situation made me wonder if there is a random reed, but I could not find it in the codes.

NOTE: I don't use CV. I am working on time series.

## Reproducible example

```python
########################
# Final Model
########################
feature_cols = aaa['feature'].values.tolist()  # LightGBM parameters

Y_train = train['quantity']
X_train = train[feature_cols]

Y_val = val['quantity']
X_val = val[feature_cols]

lgb_params = {'metric': 'mae',
              'num_leaves': 10,
              'learning_rate': 0.02,
              'feature_fraction': 0.8,
              'max_depth': 5,
              'verbose': -1,
              'num_boost_round': 12000,
              'early_stopping_rounds': 200,
              'nthread': -1}

# metric mae: l1, absolute loss, mean_absolute_error, regression_l1
# l2, square loss, mean_squared_error, mse, regression_l2, regression
# rmse, root square loss, root_mean_squared_error, l2_root
# mape, MAPE loss, mean_absolute_percentage_error


lgbtrain = lgb.Dataset(data=X_train, label=Y_train, feature_name=cols)
lgbval = lgb.Dataset(data=X_val, label=Y_val, reference=lgbtrain, feature_name=cols)

model = lgb.train(lgb_params, lgbtrain,
                  valid_sets=[lgbtrain, lgbval],
                  num_boost_round=lgb_params['num_boost_round'],
                  early_stopping_rounds=lgb_params['early_stopping_rounds'],
                  verbose_eval=3000)
```

## Environment info

LightGBM version or commit hash: 3.2.1

Command(s) you used to install LightGBM

```shell
pip install lightgbm
```

## Additional Comments

now my code and my modeling result show no improvement and I can't get results
I don't use sklearn API using your base
